Game/Source/GDKTestGyms/Private/NFRConstants.cpp

```diff
@@ -7,25 +7,32 @@
 
 DEFINE_LOG_CATEGORY(LogNFRConstants);
 
-bool UNFRConstants::SamplesForFPSValid() const
+
+FMetricDelay::FMetricDelay()
+	: TimeToStart(0)
+{
+}
+
+FMetricDelay::FMetricDelay(int64 InTimeToStart)
+	: TimeToStart(InTimeToStart)
 {
-	checkf(bIsInitialised, TEXT("NFRConstants not initialised"));
-	if (bFPSSamplingValid)
-	{
-		return true;
-	}
-	bool bIsValidNow = FDateTime::Now().GetTicks() > TimeToStartFPSSampling;
-	if (bIsValidNow)
-	{
-		bFPSSamplingValid = true;
-	}
-	return bIsValidNow;
 }
 
-void UNFRConstants::InitWithWorld(UWorld* World)
+bool FMetricDelay::IsReady() const
 {
-	TimeToStartFPSSampling = FDateTime::Now().GetTicks() + FTimespan::FromSeconds(120.0f).GetTicks();
+	return FDateTime::Now().GetTicks() > TimeToStart;
+}
 
+UNFRConstants::UNFRConstants()
+	: PlayerCheckMetricDelay(FDateTime::Now().GetTicks() + FTimespan::FromSeconds(15.0f * 60.0f).GetTicks())
```

Comment:
Just pass in the delta, and let the Delay object calculate the offset.

Reply:
TRU!